go/vt/vtadmin/api.go

```diff
@@ -271,7 +272,7 @@ func (api *API) WithCluster(c *cluster.Cluster, id string) dynamic.API {
 	}
 
 	if c != nil {
-		if existingCluster, exists := api.clusterMap[id]; !exists || existingCluster != c {
+		if existingCluster, exists := api.clusterMap[id]; !exists || !reflect.DeepEqual(existingCluster, c) {
```

Comment:
ah, so to clarify: i don't think we should use reflection, i think we should either be okay with pointer equality OR define our own Equal method and decide on what _we_ mean for two clusters to be equivalent